### DISCUSSION TITLE
How to use model in computed decorator

### DISCUSSION THREAD

**Discussion by binsarjr** [adonisjs/core#2883] [Help] [answered]
how to use model in computed decorator. i want to check one data against other data.

when use laravel it's more or less like this
```php
public function getHasParentAttribute()
{
    if($this->parent == '0'){
        return false;
    }
    $check = FAQ::where('id', $this->parent)->first();
    if($check) {
        return true;
    }
    return false;
}
```

**Comment by feniljariwala82** [comment, marked as answer]
**NOTE**: You can use the computed decorator to serialize a model property when **converting** the model instance **to a JSON object.**

Here is an example of how to using "@computed()" decorator. I am simply converting my detail object to a JSON object.

import { computed } from '@ioc:Adonis/Lucid/Orm'


// this should be written in your model only. 
 @computed()
// method name can be anything, but different from model methods!
  public get parsedDetail() {
    const parsedDetail: any = JSON.parse(this.detail)
    return parsedDetail
  }

I hope this helps.

**Reply by binsarjr** [reply]
I don't mean that. But, how to do a query inside `@computed`. While to do a query you have to use async/await and `@computed` does not support that.

actually I've found the solution. I forgot to mention, thank you for making me remember.

i use hook+raw query method.
```js
@beforeFind()
@beforeFetch()
public static neededQuery(query: ModelQueryBuilderContract<typeof Faq>) {
  query.select(
    '*',
    Database.raw(
      `CASE
        WHEN parent != 0 THEN true
        WHEN
          (select id from pertanyaan p2 where p2.id=${Faq.table}.parent) != null
          THEN true
      ELSE false END has_parent`
    ),
    Database.raw(
      `
      CASE
        WHEN
          (select count(*) from pertanyaan p2 where p2.parent=${Faq.table}.id) != 0
          THEN true
      ELSE false END has_child`
    )
  )
}
```

I hope this helps someone else

**Reply by feniljariwala82** [reply]
Well, there is a way to do that.

In your **neededQuery** function, you can write async/await this way, 
public static neededQuery(query: ModelQueryBuilderContract<typeof Faq>) {
(async function () {
      try {
        console.log('success')
      } catch (error) {
        console.log('error')
      }
    })()
}

This is one way of using async/await. You may try this out.
